reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java

```diff
@@ -0,0 +1,340 @@
+		Disposable d = test.schedulePeriodically(
+			() -> {
+				try {
+					Thread.sleep(500);
+				}
+				catch (InterruptedException e) {
+					throw new RuntimeException(e);
+				}
+				finally {
+					latch.countDown();
+				}
+			},
+			100, 100, TimeUnit.MILLISECONDS);
+		latch.await(10, TimeUnit.SECONDS);
+		d.dispose();
+
+		//now we assert that the completedTasks timer reflects an history of all Runnable#run
+		assertThat(test.submittedDirect.count()).as("#submittedDirect").isZero();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("#submittedPeriodically").isOne();
+		assertThat(test.completedTasks.count())
+			.as("#completed")
+			.isEqualTo(3L);
+		assertThat(test.completedTasks.totalTime(TimeUnit.MILLISECONDS))
+			.as("total duration of tasks")
+			.isCloseTo(1500, Offset.offset(200d));
+	}
+
+	@Test
+	void scheduleIncrementDirectCounterOnly() {
+		TimedScheduler test = new TimedScheduler(Schedulers.immediate(), registry, "test", Tags.empty());
+
+		test.schedule(() -> {});
+
+		assertThat(test.submittedDirect.count()).as("submittedDirect.count").isOne();
+		assertThat(test.submittedDelayed.count()).as("submittedDelayed.count").isZero();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("submittedPeriodicallyInitial.count").isZero();
+		assertThat(test.submittedPeriodicallyIteration.count()).as("submittedPeriodicallyIteration.count").isZero();
+	}
+
+	@Test
+	void scheduleDelayIncrementsDelayedCounter() throws InterruptedException {
+		TimedScheduler test = new TimedScheduler(Schedulers.single(), registry, "test", Tags.empty());
+
+		test.schedule(() -> {}, 100, TimeUnit.MILLISECONDS);
+
+		assertThat(test.submittedDirect.count()).as("submittedDirect.count").isZero();
+		assertThat(test.submittedDelayed.count()).as("submittedDelayed.count").isOne();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("submittedPeriodicallyInitial.count").isZero();
+		assertThat(test.submittedPeriodicallyIteration.count()).as("submittedPeriodicallyIteration.count").isZero();
+	}
+
+	@Test
+	void schedulePeriodicallyIncrementsPeriodicallyCountersInitialByOneAndIterationByFour() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(5);
+		TimedScheduler test = new TimedScheduler(Schedulers.single(), registry, "test", Tags.empty());
+
+		Disposable d = test.schedulePeriodically(latch::countDown,100, 100, TimeUnit.MILLISECONDS);
+
+		latch.await(10, TimeUnit.SECONDS);
+		d.dispose();
+
+		assertThat(test.submittedDirect.count()).as("submittedDirect.count").isZero();
+		assertThat(test.submittedDelayed.count()).as("submittedDelayed.count").isZero();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("submittedPeriodicallyInitial.count").isOne();
+		assertThat(test.submittedPeriodicallyIteration.count()).as("submittedPeriodicallyIteration.count").isEqualTo(4);
+	}
+
+	@Test
+	void schedulePeriodicallyIncrementsCompletedTasksByNumberOfIterations() throws InterruptedException {
```

Comment:
Can this test and the previous be combined to one? If the name was more generic, like `schedulePeriodicallyIsProperlyMetered"? In that case it's easier to see all the expectations we have when a periodic task is scheduled.

Reply:
yeah that makes sense